sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorConnection.java

```diff
@@ -138,12 +138,11 @@ public ReactorConnection(String connectionId, ConnectionOptions connectionOption
                 "connectionId[%s]: Connection is disposed. Cannot get CBS node.", connectionId))));
         }
 
-        final Mono<ClaimsBasedSecurityNode> cbsNodeMono =
-            RetryUtil.withRetry(getEndpointStates().takeUntil(x -> x == AmqpEndpointState.ACTIVE),
-                connectionOptions.getRetry().getTryTimeout(), retryPolicy)
-            .then(Mono.fromCallable(this::getOrCreateCBSNode));
+        final Flux<AmqpEndpointState> activeEndpointState = RetryUtil.withRetry(
+            getEndpointStates().takeUntil(x -> x == AmqpEndpointState.ACTIVE), connectionOptions.getRetry(),
+            "ReactorConnection: Retries exhausted waiting for ACTIVE endpoint state on CBS node.");
 
-        return connectionMono.then(cbsNodeMono);
+        return Mono.when(connectionMono, activeEndpointState).then(Mono.fromCallable(() -> getOrCreateCBSNode()));
```

Comment:
This change is happening because we want to run the connectionMono and activeEndpointState publishers concurrently?

Reply:
Yes. It is. When I first wrote this code, I thought it had to sequential but it doesn't. :)